lib/Doctrine/DBAL/Platforms/PostgreSQL100Platform.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Doctrine\DBAL\Platforms;
+
+use Doctrine\DBAL\Platforms\Keywords\PostgreSQL100Keywords;
+
+/**
+ * Provides the behavior, features and SQL dialect of the PostgreSQL 10.0 database platform.
+ *
+ * @link   www.doctrine-project.org
+ * @since  2.6
+ */
+class PostgreSQL100Platform extends PostgreSQL94Platform
+{
+	/**
+	 * {@inheritdoc}
+	 */
+	protected function getReservedKeywordsClass() : string
+	{
+		return PostgreSQL100Keywords::class;
+	}
+
+	public function getSequenceMetadataSQL(string $rawSequenceName, string $schemaName) : string
+	{
+		$sequenceName = $this->cleanSequenceNameFromSchemaName($rawSequenceName, $schemaName);
+
+		return 'SELECT min_value, increment_by FROM pg_sequences WHERE schemaname = '.$this->quoteStringLiteral($schemaName).' AND sequencename = ' . $this->quoteStringLiteral($sequenceName);
```

Comment:
Given that in PostgreSQL 10, the sequence metadata is available via a view, would it make sense to rework `getListSequencesSQL()` instead? With this approach, the client will not have to execute the SQL from  `getSequenceMetadataSQL()` for each sequence, and the method itself will not need to be introduced on the abstract platform level.

Reply:
Not sure it's a good idea. `getListSequencesSQL()` lists sequences. It would need to add also `increment_by` and `min_value` columns. That would change the method's functionality. This method is called before in `listSequences()` already.

Reply:
Look at the `OraclePlatform`. Its `getListSequencesSQL()` returns list and metadata. The only reason it doesn't do the same for PostgreSQL is that prior to version 10 there was no view to fetch the metadata from.